apps/server/src/modules/tldraw/service/tldraw.ws.service.spec.ts

```diff
@@ -0,0 +1,376 @@
+		expect(service.pingTimeout).toBeDefined();
+		expect(service.persistence).toBeDefined();
+	});
+
+	describe('when client is not connected to WS', () => {
+		const setup = async () => {
+			ws = await TestHelper.setupWs(wsUrl);
+
+			const closeConSpy = jest.spyOn(service, 'closeConn').mockImplementationOnce(() => {});
+			const sendSpy = jest.spyOn(service, 'send');
+			const doc: { conns: Map<WebSocket, Set<number>> } = { conns: new Map() };
+			const byteArray = new TextEncoder().encode(testMessage);
+
+			return {
+				closeConSpy,
+				sendSpy,
+				doc,
+				byteArray,
+			};
+		};
+
+		it('should throw error for send message', async () => {
+			const { closeConSpy, sendSpy, doc, byteArray } = await setup();
+
+			service.send(doc as WsSharedDocDo, ws, byteArray);
+
+			expect(sendSpy).toThrow();
+			expect(sendSpy).toHaveBeenCalledWith(doc, ws, byteArray);
+			expect(closeConSpy).toHaveBeenCalled();
+
+			ws.close();
+			sendSpy.mockRestore();
+		});
+	});
+
+	describe('when websocket has ready state different than 0 or 1', () => {
+		const setup = () => {
+			const closeConSpy = jest.spyOn(service, 'closeConn');
+			const sendSpy = jest.spyOn(service, 'send');
+			const doc: { conns: Map<WebSocket, Set<number>> } = { conns: new Map() };
+			const socketMock = { readyState: 3, close: () => {} };
+			const byteArray = new TextEncoder().encode(testMessage);
+
+			return {
+				closeConSpy,
+				sendSpy,
+				doc,
+				socketMock,
+				byteArray,
+			};
+		};
+
+		it('should close connection if websocket has ready state different than 0 or 1', () => {
+			const { closeConSpy, sendSpy, doc, socketMock, byteArray } = setup();
+
+			service.send(doc as WsSharedDocDo, socketMock as WebSocket, byteArray);
+
+			expect(sendSpy).toHaveBeenCalledWith(doc, socketMock, byteArray);
+			expect(sendSpy).toHaveBeenCalledTimes(1);
+			expect(closeConSpy).toHaveBeenCalled();
+
+			closeConSpy.mockRestore();
+			sendSpy.mockRestore();
+		});
+	});
+
+	describe('when websocket has ready state 0', () => {
+		const setup = async () => {
+			ws = await TestHelper.setupWs(wsUrl);
+
+			const sendSpy = jest.spyOn(service, 'send');
+			const doc: { conns: Map<WebSocket, Set<number>> } = { conns: new Map() };
+			const socketMock = { readyState: 0, close: () => {} };
+			doc.conns.set(socketMock as WebSocket, new Set());
+			const encoder = encoding.createEncoder();
+			encoding.writeVarUint(encoder, 2);
+			const updateByteArray = new TextEncoder().encode(testMessage);
+			encoding.writeVarUint8Array(encoder, updateByteArray);
+			const msg = encoding.toUint8Array(encoder);
+			return {
+				sendSpy,
+				doc,
+				msg,
+			};
+		};
+
+		it('should call send in updateHandler', async () => {
+			const { sendSpy, doc, msg } = await setup();
+
+			service.updateHandler(msg, {}, doc as WsSharedDocDo);
+
+			expect(sendSpy).toHaveBeenCalled();
+
+			ws.close();
+			sendSpy.mockRestore();
+		});
+	});
+
+	describe('when received message of specific type', () => {
```

Comment:
now we have a whole set of describe blocks that are about messageHander, these should be grouped in a describe('messageHandler')

Reply:
Added suggested describe